ui/src/containers/GphlWorkflowParametersDialog.jsx

```diff
@@ -1,62 +1,396 @@
-import React from 'react';
+/* eslint-disable unicorn/prefer-query-selector */
+import React, { useState, useEffect, useCallback } from 'react';
 import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
-import { Modal } from 'react-bootstrap';
-import Form from '@rjsf/core';
-import validator from '@rjsf/validator-ajv8';
+import { Modal, Row, Col, Form, Table, Button, Stack } from 'react-bootstrap';
+import './WorkflowParametersDialog.css';
+
 import {
   showGphlWorkflowParametersDialog,
-  submitWorkflowParameters,
+  updateGphlWorkflowParameters,
+  updateGphlWorkflowParametersDialog,
 } from '../actions/workflow';
 
-import './WorkflowParametersDialog.css';
+const uiOptions = 'ui:options';
+
+function renderIndexingTable(indexingTable, selected, onSelectRow) {
+  return (
+    <Table id="indexing_table" bordered responsive className="indexing_table">
+      <thead>
+        <tr>
+          <th> </th>
+          {indexingTable.header.map((thContent) => (
+            <th
+              key={thContent}
+              className="indexing_table_special_td_th text-center"
+            >
+              <pre>{thContent}</pre>
+            </th>
+          ))}
+          <th className="indexing_table_special_td_th"> </th>
+        </tr>
+      </thead>
+      <tbody className="indexing_table_body">
+        {indexingTable.content.map((tdContents) =>
+          tdContents.map((tdContent, index) => (
+            <tr
+              key={tdContent}
+              className={`${
+                selected.includes(index) ? 'indexing_table_row_selected' : ''
+              }
+                  ${
+                    indexingTable.highlights[index + 1]
+                      ? indexingTable.highlights[index + 1][0]
+                      : ''
+                  } trclass`}
+              onClick={() => onSelectRow(index, tdContent)}
```

Comment:
Would be good to make this a bit more accessible. I don't think you could interact with your table with the keyboard, for instance. A column with a native checkbox would solve this. I'm sure you could find a bunch of good examples of accessible tables with selectable rows online. You could even use a headless UI library such as React Aria: https://react-spectrum.adobe.com/react-aria/Table.html

Reply:
Sure, I prefer using a custom table for now, cause in the future the uiSchema may become more complex where you'll probably need to select column  instead of Row. 

That's a good one by the way, i did not know about it : react-aria-components

Reply:
The current way the system works (as coded for Qt) you actually select and highlight neither columns, nor rows but *cells*. Since the only current use is for a single-column table, it is no problem if you can only select rows - for now - but selecting cells is the most likely evolution. Of course we might want even more different or complex widgets at some point, but those would have to be coded when we need them.